lib/src/Uncertainty/Algorithm/Analytical/MultiFORM.cxx

```diff
@@ -0,0 +1,233 @@
+  */
+MultiFORM::MultiFORM(const OptimizationAlgorithm & solver,
+                      const Event & event,
+                      const Point & physicalStartingPoint)
+: FORM(solver, event, physicalStartingPoint)
+, maximumNumberOfDesignPoints_(ResourceMap::GetAsUnsignedInteger("MultiFORM-DefaultMaximumNumberOfDesignPoints"))
+{
+  // Nothing to do
+}
+
+/* Virtual constructor */
+MultiFORM * MultiFORM::clone() const
+{
+  return new MultiFORM(*this);
+}
+
+/* Result accessor */
+MultiFORMResult MultiFORM::getResult() const
+{
+  return result_;
+}
+
+
+/* String converter */
+String MultiFORM::__repr__() const
+{
+  OSS oss;
+  oss << "class=" << MultiFORM::GetClassName()
+      << " derived from" << FORM::__repr__()
+      << " result=" << result_;
+  return oss;
+}
+
+
+/* Function that computes the design point by re-using the Analytical::run() and creates a MultiFORMResult */
+void MultiFORM::run()
+{
+  const Scalar gamma = ResourceMap::GetAsScalar("MultiFORM-DefaultGamma");
+  const Scalar delta = ResourceMap::GetAsScalar("MultiFORM-DefaultDelta");
+  const Scalar gamma2 = gamma * gamma;
+  const Scalar delta2 = delta * delta;
+  const UnsignedInteger dimension = getEvent().getImplementation()->getAntecedent().getDimension();
+  const Scalar bound = acos(1.0 - 0.5 * gamma2);
+
+  Collection<Point> designPointCollection;
+  Point betaCollection;
+  UnsignedInteger numberOfDesignPointsFound = 0;
+  const StandardEvent standardEvent(getEvent());
+
+  // initialize the new limit state function
+  Point coefficients(1, 1.0);
+  Collection<Function> collection(1, standardEvent.getImplementation()->getFunction());
+
+  Collection<FORMResult> formResultCollection;
+  Bool designPointValidity = true;
+  while (designPointValidity)
+  {
+    OptimizationAlgorithm solver(getNearestPointAlgorithm());
+    const LinearCombinationFunction levelFunction(collection, coefficients);
+    NearestPointProblem optimizationProblem(levelFunction, standardEvent.getThreshold());
+    solver.setProblem(optimizationProblem);
+    setNearestPointAlgorithm(solver);
+    try
+    {
+      FORM::run();
+    }
+    catch (Exception & ex)
+    {
+      Log::Warn(String("MultiFORM: ") + ex.what());
+      break;//then exit while-loop
+    }
+    const FORMResult formResult(FORM::getResult());
+    const Point standardSpaceDesignPoint(formResult.getStandardSpaceDesignPoint());
+
+    // retrieve some results
+    const Scalar beta = formResult.getHasoferReliabilityIndex();
+    designPointCollection.add(standardSpaceDesignPoint);
+    betaCollection.add(beta);
+
+    for (UnsignedInteger i = 0; i < numberOfDesignPointsFound; ++ i)
+    {
+      for (UnsignedInteger j = i + 1; j <= numberOfDesignPointsFound; ++ j)
+      {
+        Scalar v = designPointCollection[i].dot(designPointCollection[j]);
+        if (acos(v / (betaCollection[i] * betaCollection[j])) <= bound)
+        {
+          designPointValidity = false;
```

Comment:
BTW the test should be replaced by `designPointValidity = (v / (betaCollection[i] * betaCollection[j])>=bound);`